core/test/gtest/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+function(add_gtest_main suffix definitions)
+    add_library(ginkgo_gtest_main${suffix} STATIC ginkgo_main.cpp resources.cpp)
```

Comment:
any reason for STATIC?

Reply:
Windows seems to have issues with global variables being declared in a shared library, but referenced from another library. This seems the easiest fix